### PR TITLE
[Bug Fix] utils/url.js removing trailing slashes

### LINKED ISSUE #19207: Bug in `utils/url.js` removing trailing slashes
### Describe the Bug

The [utility `/api/src/utils/url.ts`](https://github.com/directus/directus/blob/c2c04734f0c5e1fa72090c5ca624fcb32d4cf3c9/api/src/utils/url.ts#L3) removes trailing slashes.

### Why is it a problem?

Because static deployments on S3 redirect folders and forces a trailing slash. But when it does so, it removes the query line. Dumb, I know! This issue is well documented, see [here](https://github.com/gatsby-uc/gatsby-plugin-s3/issues/51), [here](https://repost.aws/knowledge-center/s3-static-website-url-trailing-slash) and [here](https://stackoverflow.com/questions/30090254/aws-s3-forces-302-redirects-when-url-has-no-trailing-slash-need-301s).

From [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/IndexDocumentSupport.html#IndexDocumentsandFolders):
> However, if you exclude the trailing slash from the preceding URL, Amazon S3 first looks for an object `photos` in the bucket. If the `photos` object is not found, it searches for an index document, `photos/index.html`. If that document is found, Amazon S3 returns a `302 Found` message and points to the `photos/` key. For subsequent requests to `photos/`, Amazon S3 returns `photos/index.html`. If the index document is not found, Amazon S3 returns an error.

For more context, I use Nuxt 3 in my project, and it creates paths like `dist/verify/index.html` when using static site generation mode.

### To Reproduce

I give it the URL `https://example.com/verify/` and it returns `https://example.com/verify?token=...`.  
**Notice that the utility removes the trailing slash.**

You can test using this snippet:
```js
import { Url } from "@directus/api/utils/url";

const inviteURL = new Url("http://localhost/auth/verify/");
inviteURL.setQuery("token", "123123");
console.log("inviteURL", inviteURL.toString()); 
```

This will return `http://localhost/auth/verify?token=123123` instead of `http://localhost/auth/verify/?token=123123`, which is missing the trailing slash `/`.

### Directus Version

v10.4.3

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by amerkay**
Hi!

TLDR; `Url("https://example.com/verify/").toString()` returns `https://example.com/verify`, it removes the trailing slash. It should preserve it. This PR fixes that.

- PR closes this issue https://github.com/directus/directus/issues/19207
- Tests added.

Cheers!

Fixes #19207

### CODE REVIEW

File: .changeset/amazing-flying-monkey.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": patch
+---
+
+[Bug Fix] utils/url.js removing trailing slashes. It now preserves them.
```
Comment (paescuj) [inline .changeset/amazing-flying-monkey.md:5]:
```suggestion
Fixed URL util to preserve trailing slashes
```

**Review by paescuj — APPROVED** [review]
❤️

**Review by licitdev — APPROVED** [review]
@amerkay Thanks for the PR! 👍
